symfony/plugins/orangehrmTimePlugin/Api/TimesheetAPI.php

```diff
@@ -0,0 +1,113 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Time\Api;
+
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Entity\Timesheet;
+use OrangeHRM\Time\Api\Model\TimesheetModel;
+use OrangeHRM\Time\Traits\Service\TimesheetServiceTrait;
+
+class TimesheetAPI extends Endpoint implements CollectionEndpoint
+{
+    use AuthUserTrait;
+    use TimesheetServiceTrait;
+    use DateTimeHelperTrait;
+
+    public const PARAMETER_DATE = 'date';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        $timesheet = new Timesheet();
+        $empNumber = $this->getAuthUser()->getEmpNumber();
+        $startDate = $this->getRequestParams()->getDateTime(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_DATE);
+        $timesheet->setEmployeeId($empNumber);
+        $this->getTimesheetService()->saveTimesheet($timesheet, $startDate);
+        return new EndpointResourceResult(TimesheetModel::class, $timesheet);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_DATE,
+                    new Rule(Rules::API_DATE),
+                    new Rule(Rules::CALLBACK, [
+                        function () {
+                            $date = $this->getRequestParams()->getDateTime(
+                                RequestParams::PARAM_TYPE_BODY,
+                                self::PARAMETER_DATE
+                            );
+                            return !($date > $this->getDateTimeHelper()->getNow());
+                        }
+                    ])
```

Comment:
Better to handle if timesheet has already created

Reply:
done